qiskit_ibm/ibm_provider.py

```diff
@@ -0,0 +1,965 @@
+    specified name::
+
+        simulator_backend = provider.get_backend('ibmq_qasm_simulator')
+
+    It is also possible to use the ``backend`` attribute to reference a backend.
+    As an example, to retrieve the same backend from the example above::
+
+        simulator_backend = provider.backend.ibmq_qasm_simulator
+
+    Note:
+        The ``backend`` attribute can be used to autocomplete the names of
+        backends available to this provider. To autocomplete, press ``tab``
+        after ``provider.backend.``. This feature may not be available
+        if an error occurs during backend discovery. Also note that
+        this feature is only available in interactive sessions, such as
+        in Jupyter Notebook and the Python interpreter.
+    """
+
+    _providers = OrderedDict()  # type: Dict[HubGroupProject, IBMProvider]
+    _account = None  # type: Dict
+
+    def __new__(
+            cls,
+            token: Optional[str] = None,
+            url: Optional[str] = None,
+            hub: Optional[str] = None,
+            group: Optional[str] = None,
+            project: Optional[str] = None,
+            **kwargs: Any
+    ) -> 'IBMProvider':
+        """IBMProvider constructor
+
+        Args:
+            token: IBM Quantum token.
+            url: URL for the IBM Quantum authentication server.
+            hub: Name of the hub to use.
+            group: Name of the group to use.
+            project: Name of the project to use.
+            **kwargs: Additional settings for the connection:
+
+                * proxies (dict): proxy configuration.
+                * verify (bool): verify the server's TLS certificate.
+
+        Returns:
+            If `hub`, `group`, and `project` are specified, the corresponding provider
+            is returned. Otherwise the provider for the open access project is returned.
+
+        Raises:
+            IBMProviderCredentialsInvalidFormat: If the default provider saved on
+                disk could not be parsed.
+            IBMProviderCredentialsNotFound: If no IBM Quantum credentials
+                can be found.
+            IBMProviderCredentialsInvalidUrl: If the URL specified is not
+                a valid IBM Quantum authentication URL.
+            IBMProviderCredentialsInvalidToken: If the `token` is not a valid
+                IBM Quantum token.
+        """
+        account_credentials, account_preferences, hub, group, project = cls._resolve_credentials(
+            token=token,
+            url=url,
+            hub=hub,
+            group=group,
+            project=project,
+            **kwargs
+        )
+        account = cls._account
+        if not account and (not cls._providers or
+                            cls._is_different_account(account_credentials.token)):
+            cls._initialize_providers(credentials=account_credentials,
+                                      preferences=account_preferences)
+            cls._account = None
```

Comment:
I think you're using this `cls._account` to decide whether to "skip" `__new__` processing to avoid infinite loop. If that's the case, it can really use some comments because it's not obvious at all (especially with the variable name `account`).

Reply:
Fixed!